src/util/antoraSupport.ts

```diff
@@ -0,0 +1,36 @@
+import vscode from 'vscode'
+import fs from 'fs'
+import yaml from 'js-yaml'
+import path from 'path'
+
+export default class AntoraSupport {
+  public async isDetected (): Promise<boolean> {
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 10)
+    return antoraYmlFiles.length > 0
+  }
+
+  public async findModulesDirPath (textDocument: vscode.TextDocument): Promise<any> {
+    const pathToAsciidocFile = textDocument.uri.fsPath
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 100)
+    // check if antora.yml file matches with textDocument content source root
+    try {
+      for (let i = 0; i < antoraYmlFiles.length; i++) {
+        if (pathToAsciidocFile.startsWith(path.dirname(antoraYmlFiles[i].path))) {
+          return pathToAsciidocFile.substring(0, pathToAsciidocFile.indexOf('modules'))
+        }
+      }
+    } catch (err) {
+      await vscode.window.showErrorMessage('Antora modules is out of the current workspace, ' + err.toString())
+    }
+  }
+
+  public async getAttributes (textDocument: vscode.TextDocument): Promise<{ [key: string]: string }> {
+    try {
+      const doc = yaml.load(fs.readFileSync(await this.findModulesDirPath(textDocument) + 'antora.yml', 'utf8'))
+      return doc.asciidoc.attributes
+    } catch (err) {
+      await vscode.window.showErrorMessage('Error parsing antora.yml, ' + err.toString())
```

Comment:
Again, I don't think we should show an error message. It's not a big deal if the autocompletion does not show any attributes from `antora.yml` if the file is not properly formatted as YAML.

In my opinion, using `console.log` with the full path to the `antora.yml` file is enough:

```js
console.log(`Unable to parse /path/to/docs/antora.yml, cause: err.toString()`)
```